src/bitcoinrpc.cpp

```diff
@@ -174,40 +181,70 @@ string CRPCTable::help(string strCommand) const
         {
             // Help text is returned in an exception
             string strHelp = string(e.what());
-            if (strCommand == "")
+            if (strCommand.empty())
                 if (strHelp.find('\n') != string::npos)
                     strHelp = strHelp.substr(0, strHelp.find('\n'));
             strRet += strHelp + "\n";
         }
     }
-    if (strRet == "")
+    if (strRet.empty())
         strRet = strprintf("help: unknown command: %s\n", strCommand);
     strRet = strRet.substr(0,strRet.size()-1);
     return strRet;
 }
 
 Value help(const Array& params, bool fHelp)
 {
-    if (fHelp || params.size() > 1)
+    if (fHelp || params.size() == 0 || params.size() > 1)
         throw runtime_error(
-            "help [command]\n"
-            "List commands, or get help for a command.");
-
+            "help [command/category]\n"
+            "Returns help on a specific command or category you request\n"
+            "\n"
+            "help command --> Returns help for specified command; ex. help backupwallet\n"
+            "\n"
+            "Categories:\n"
+            "wallet --------> Returns help for blockchain related commands\n"
+            "mining --------> Returns help for neural network/cpid/beacon related commands\n"
+            "developer -----> Returns help for developer commands\n"
+            "network -------> Returns help for network related commands\n");
```

Comment:
This returns with (code -1) and in red in the Wallet rpcconsole

Reply:
coudl be i'll check and i wouldnt be surprised as the merge conflicts from a horror movies i had to work out lol.

Reply:
ya categories work i just got to fix the pass through for help from the commands. which was fine before merging recent changes. no problems. will do it later today

Reply:
i can confirm that `help` works `help wallet` `help mining` `help developer` and `help network` works but `help commandshere` does not however that will be a easy fix i remember i didnt save the fix for that for some reason.